operations/helm/charts/mimir-distributed/templates/validate.yaml

```
@@ -23,3 +23,25 @@
 {{- if not (has .Values.configStorageType (list "Secret" "ConfigMap")) }}
 {{- fail "The setting 'configStorageType' should be one of 'Secret' or 'ConfigMap'." }}
 {{- end }}
+
+{{- with .Values.metaMonitoring.grafanaAgent }}
+{{/*
+  !servmon.Enabled
+  && agent.Enabled
+  && (remote.url != "" || (len(additionalConfigs) > 0 && additionalConfigs.url != ""))
+*/}}
+{{- if and
+    (not (($.Values).serviceMonitor).enabled)
+    (and
+      .enabled
+      (or
+        (not (empty ((.metrics).remote).url))
+        (and
+          (not (empty (.metrics).additionalRemoteWriteConfigs))
+          (not (empty (first (.metrics).additionalRemoteWriteConfigs).url))
+        )
+      )
+    )  }}
```

not: a little simplification as `and` can take multiple arguments

```suggestion
{{- if and
    (not (($.Values).serviceMonitor).enabled)
    .enabled
    (or
      (not (empty ((.metrics).remote).url))
      (and
        (not (empty (.metrics).additionalRemoteWriteConfigs))
        (not (empty (first (.metrics).additionalRemoteWriteConfigs).url))
      )
    )
    }}
```